src/SimplePromiseQueue.ts

```diff
@@ -0,0 +1,24 @@
+export default class SimplePromiseQueue {
+    private readonly _queue: Promise<void>[] = [];
```

Comment:
Can you use Array<Promise<void>> instead ? (stickler for Typescript non-simple-types) :P

Reply:
And no semicolon please